builtin/gc.c

```diff
@@ -1678,6 +1657,9 @@ static const char *get_frequency(enum schedule_priority schedule)
 	}
 }
 
+static const char *custom_background_config =
+	"-c credential.interactive=false -c cred.askPass=false";
```

Comment:
I wonder if it's worth trying to upstream the concept of `credential.interactive`, and have it mean "helpers should not interact with the user".

Right now, this is a GCM specific thing.

Alternatively, perhaps a `core.interactive` or `core.background` setting would make more sense? To indicate that "this instance of Git is running non-interactively or in the background", and that is a signal that helpers can pick up on to mean "no prompt"?